cdap-ui/app/directives/view-queries/view-queries.js

```diff
@@ -28,28 +33,42 @@ angular.module(PKG.name + '.commons')
         $scope.responses = {};
 
         $scope.fetchResult = function(query) {
-          $scope.responses.request = query;
+          // Close other accordion
+          angular.forEach($scope.queries, function(q) {
+            q.isOpen = false;
+          });
 
-          // request schema
-          dataSrc
-            .request({
-              _cdapPath: '/data/explore/queries/' +
-                            query.query_handle + '/schema'
-            })
-            .then(function (result) {
-              $scope.responses.schema = result;
-            });
+          query.isOpen = !query.isOpen;
+
+          if (query.isOpen) {
+            $scope.responses.request = query;
+
+            // request schema
+            dataSrc
+              .request({
+                _cdapPath: '/data/explore/queries/' +
+                              query.query_handle + '/schema'
+              })
+              .then(function (result) {
+                angular.forEach(result, function(v) {
+                  v.name = v.name.split('.')[1];
+                });
+
+                $scope.responses.schema = result;
+              });
+
+            // request preview
+            dataSrc
+              .request({
+                _cdapPath: '/data/explore/queries/' +
+                              query.query_handle + '/preview',
+                method: 'POST'
+              })
+              .then(function (result) {
```

Comment:
Not necessarily a bad thing but we already have 'result' being fetched (/schema). This is a preview of the results. Thought it would be better if this is 'results'. Feel too many results are there in this controller :smile: